pipeline_dp/pipeline_operations.py

```diff
@@ -42,7 +43,8 @@ def filter_by_key(self, col, public_partitions, stage_name: str):
 
         Args:
           col: collection with elements (partition_key, data).
-          public_partitions: collection of public partition keys.
+          public_partitions: collection of public partition keys,
```

Comment:
please rename `public_partitions` to `keys_to_keep` and in other `filter_by_key` implementations as well.

**Explanation**: we try to keep `PipelineOperations` are focused on pipeline operations not to DP. I missed that naming in previous code reviews.

Reply:
changed